Support and run analysis on CI

JohnnyMorganz opened the pull request:
This PR:
- Introduces require support to the `lute --check` command. This is copied from `luau-analyze`
- Creates a set of definition files in a definition folders for C-based apis under `@lute`. This alias is registered in the `.luaurc` file
- Adds a CI check to run analysis on std, batteries and examples

A couple points:
- Decided to only support new solver (`LuauSolverV2` is force-enabled) for dogfooding purposes.

Still need to fix some type errors

JohnnyMorganz committed:
Run analysis on CI

aatxe commented on vm/include/lute/vm.h:
```diff
@@ -16,7 +16,7 @@ namespace vm
 int lua_defer(lua_State* L);
 
 static const luaL_Reg lib[] = {
-    {"create", lua_spawn},
```
This was actually intentional. I wanted consistency with other `create` constructors for builtins in Luau.

aatxe commented on definitions/vm.luau:
```diff
@@ -0,0 +1,7 @@
+local vm = {}
+
+function vm.spawn(file: string): any
```
```suggestion
function vm.create(file: string): any
```

aatxe commented on vm/include/lute/vm.h:
```diff
@@ -16,7 +16,7 @@ namespace vm
 
 static const luaL_Reg lib[] = {
-    {"create", lua_spawn},
+    {"spawn", lua_spawn},
```
```suggestion
    {"create", lua_spawn},
```

aatxe reviewed (changes requested):
Looks good to me, besides the nit about vm.create

aatxe commented:
Looks like the CI setup isn't quite right though?

JohnnyMorganz committed:
Merge branch 'primary' of https://github.com/aatxe/lute into run-analysis

JohnnyMorganz committed:
Switch to vm.create

Co-authored-by: ariel <aweiss@hey.com>

JohnnyMorganz committed:
Fix examples for vm.create

JohnnyMorganz committed:
Fix type errors in json.luau

Some luau bugs, but also Luau doesn't like the forward
declared `serialize_any` and `deserialize` functions using
a local, so we switch it to a normal function.

JohnnyMorganz commented:
> Looks like the CI setup isn't quite right though?

Was waiting on the other PRs. A rebase fixed it now, but we do get a lot of type errors. Will try and fix some of them.

aatxe committed:
Merge branch 'primary' into run-analysis

aatxe committed:
Merge branch 'primary' into run-analysis

vrn-sn commented:
Saw this PR was closed. Are we good to delete the branch?

JohnnyMorganz commented:
Yeah should be fine (would be nice to get analyze running in CI someday)
